core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/DatabaseUpgradeRuntimeValidator.java

```diff
@@ -0,0 +1,42 @@
+package com.sequenceiq.cloudbreak.service.upgrade.rds;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.util.VersionComparator;
+
+@Component
+public class DatabaseUpgradeRuntimeValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseUpgradeRuntimeValidator.class);
+
+    private final VersionComparator versionComparator = new VersionComparator();
+
+    @Value("${cb.db.override.minRuntimeVersion}")
+    private String minRuntimeVersion;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    public Optional<String> findRuntimeVersionValidationError(String runtimeVersion, String accountId) {
+        Optional<String> validationFailureMessage = Optional.empty();
+        if (entitlementService.isPostgresUpgradeExceptionEnabled(accountId)) {
+            LOGGER.debug("Enable database upgrade for runtime {} because it's enabled by entitlement.", runtimeVersion);
+        } else if (0 <= versionComparator.compare(() -> runtimeVersion, () -> minRuntimeVersion)) {
```

Comment:
I would add a null check for runtime version as in some edge case we might don't have it. just to be sure